plugins/inputs/elasticsearch_query/elasticsearch_query.go

```diff
@@ -0,0 +1,310 @@
+
+    ## The date/time field in the Elasticsearch index (mandatory).
+    date_field = "@timestamp"
+
+    ## Time window to query (eg. "1m" to query documents from last minute).
+    ## Normally should be set to same as collection interval
+    query_period = "1m"
+
+    ## Lucene query to filter results
+    # filter_query = "*"
+
+    ## Fields to aggregate values (must be numeric fields)
+    # metric_fields = ["metric"]
+
+    ## Aggregation function to use on the metric fields
+    ## Must be set if 'metric_fields' is set
+    ## Valid values are: avg, sum, min, max, sum
+    # metric_function = "avg"
+
+    ## Fields to be used as tags
+    ## Must be text, non-analyzed fields. Metric aggregations are performed per tag
+    # tags = ["field.keyword", "field2.keyword"]
+
+    ## Set to true to not ignore documents when the tag(s) above are missing
+    # include_missing_tag = false
+
+    ## String value of the tag when the tag does not exist
+    ## Used when include_missing_tag is true
+    # missing_tag_value = "null"
+`
+
+// ElasticsearchQuery struct
+type ElasticsearchQuery struct {
+	URLs                []string        `toml:"urls"`
+	Username            string          `toml:"username"`
+	Password            string          `toml:"password"`
+	EnableSniffer       bool            `toml:"enable_sniffer"`
+	Timeout             config.Duration `toml:"timeout"`
+	HealthCheckInterval config.Duration `toml:"health_check_interval"`
+	Aggregations        []esAggregation `toml:"aggregation"`
+
+	Log telegraf.Logger `toml:"-"`
+
+	tls.ClientConfig
+	httpclient *http.Client
+	esClient   *elastic5.Client
+}
+
+// esAggregation struct
+type esAggregation struct {
+	Index                string          `toml:"index"`
+	MeasurementName      string          `toml:"measurement_name"`
+	DateField            string          `toml:"date_field"`
+	QueryPeriod          config.Duration `toml:"query_period"`
+	FilterQuery          string          `toml:"filter_query"`
+	MetricFields         []string        `toml:"metric_fields"`
+	MetricFunction       string          `toml:"metric_function"`
+	Tags                 []string        `toml:"tags"`
+	IncludeMissingTag    bool            `toml:"include_missing_tag"`
+	MissingTagValue      string          `toml:"missing_tag_value"`
+	mapMetricFields      map[string]string
+	aggregationQueryList []aggregationQueryData
+}
+
+// SampleConfig returns sample configuration for this plugin.
+func (e *ElasticsearchQuery) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description returns the plugin description.
+func (e *ElasticsearchQuery) Description() string {
+	return `Derive metrics from aggregating Elasticsearch query results`
+}
+
+// Init the plugin.
+func (e *ElasticsearchQuery) Init() error {
+	if e.URLs == nil {
+		return fmt.Errorf("elasticsearch urls is not defined")
+	}
+
+	err := e.connectToES()
+	if err != nil {
+		e.Log.Errorf("E! error connecting to elasticsearch: %s", err)
+		return nil
+	}
```

Comment:
That's the reason I don´t return the error here :)

Reply:
I don´t know about the Start(), I will see if it makes sense to move some logic there

Reply:
Ok, that's what I figured. I think it's fine in the current state since part of the configuration validation depends on a connection to the ES instance.

Reply:
From what I've checked, Start() applies to service input plugins only. Is that right?